util/chplenv/chpl_compiler.py

```diff
@@ -39,6 +39,60 @@ def get_prgenv_compiler():
 
     return 'none'
 
+# Figures out the compiler family (e.g. gnu) from the CC/CXX enviro vars
+# Returns '' if CC / CXX are not set and 'unknown' if they are set
+# to something too complex.
+@memoize
+def get_compiler_from_cc_cxx():
+    cc_compiler = 'unknown'
+    cxx_compiler = 'unknown'
+    warn = False
+    compiler_val = 'unknown'
+    cc_val = overrides.get('CC', '')
+    cxx_val = overrides.get('CXX', '')
+
+    if cc_val == '' and cxx_val == '':
+        return ''
+
+    if cc_val:
+        cc_compiler = get_compiler_from_command(cc_val)
+    if cxx_val:
+        cxx_compiler = get_compiler_from_command(cxx_val)
+
+    if cc_val and cxx_val:
+        if cc_compiler == cxx_compiler:
+            compiler_val = cc_compiler
+        else:
+            error("Conflicting compiler families for CC and CXX settings\n"
+                  "  {0} -> {1}\n"
+                  "  {2} -> {3}".format(cc_val, cc_val_command,
+                                        cxx_val, cxx_val_command))
```

Comment:
Looks like `cc_val_command` and `cxx_val_command` are never defined so this result in a NameError exception and not our error message

Reply:
PR #18052 will fix